pkg/types/chain_writer.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/google/uuid"
+)
+
+type ChainWriter interface {
+	// SubmitSignedTransaction packs and broadcasts a transaction to the underlying chain.
+	//
+	// The `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
+	SubmitSignedTransaction(ctx context.Context, payload []byte, signature map[string]any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) (int64, error)
+
+	// StatusForUUID returns the current status of a transaction in the underlying chain's TXM.
+	StatusForUUID(ctx context.Context, transactionID uuid.UUID) (TransactionStatus, error)
+
+	// GetFeeComponents retrieves the associated gas costs for executing a transaction.
+	GetFeeComponents() (ChainFeeComponents, error)
+}
+
+// TxMeta contains metadata fields for a transaction.
+//
+// Eventually this will replace, or be replaced by (via a move), the `TxMeta` in core:
+// https://github.com/smartcontractkit/chainlink/blob/dfc399da715f16af1fcf6441ea5fc47b71800fa1/common/txmgr/types/tx.go#L121
+type TxMeta = map[string]string
```

Comment:
There's a small issue with having this be a map of string -> string. When we decode the map to create the TxMeta, we will have to programmatically match every field name and cast each type to the appropriate types in the TxMeta struct. An alternative could be to use map of string -> any, in which case we could use JSON marshal and unmarshal to create the struct, since TxMeta is JSON encodable.

Reply:
Why isn't this a well-defined set of fields in the first place?

Reply:
@jmank88 The TxMeta struct is defined here:

https://github.com/smartcontractkit/chainlink/blob/ccb8cd85fef8e3bbe3fb5580277a7bd7f477e6bb/common/txmgr/types/tx.go#L121

We effectively want to continue to support products that want this info logged alongside their transaction, but the struct itself has a ton of generics we'd have to expose to every product using this interface. Something like I'm proposing I think lowers the burden on products that may only be using one or two fields and can pass them as map as described in the doc.

There are alternatives to this of course. We could define something that lets products know all the valid field names, but this can be gathered by importing the TxMeta struct if needed.

Reply:
If this API does not specify the expected fields, how will a product know what to pass? and how will a chain relayer know what to look for?

Reply:
(If the generics are just address and hash, then we have precedent for handling that already)

Reply:
@jmank88 can you show me the existing code for handling the generics? I just want to think about whether it makes sense to have the pass the TxMeta struct or not.

Reply:
We cannot support generics directly, but we have chain-agnostic types like the libocr `type Account string` for holding chain-specific encoded values, and we could constrain our generic types to be translatable to/from `Account` or similar.

Reply:
@jmank88 the ideal situation would be for us to move `TxMeta` from `chainlink` into `chainlink-common`, which actually is the plan long term. Ideally we'd be using the typed struct now, but we can't import `chainlink` from `chainlink-common`.

I've added a `todo` indicating the plan to do this in a future PR.

Reply:
The type you are referencing cannot be used as-is. We have to create an analogous concrete, chain-agnostic type.  IMHO now is the time to do this - I don't actually see how the piece can fit together without it.

Reply:
Is there any reason that we cannot just move (copy for this PR) the existing `TxMeta` struct **as is** from `chainlink` into `chainlink-common`, or are changes necessary?

@patrick-dowell -- thoughts on this?